File: 2025-and-before/2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int n = nums.size(),pos=0,neg = 1;
        vector<int>ans(n);
        
        for(int&i : nums){
            
            if(i<0){
                ans[neg] = i;
                neg+=2;
            }
            
            else{
                ans[pos] = i;
                pos+=2;
            }
        }
        
        return ans;
        
    }
};

File: 2025-and-before/2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {
    const sz = nums.length;
    const arr = new Array(sz);
    let pos = 0, neg =1;

    for(let i=0;i<sz;i++){
        
        if(nums[i]>=0){
            arr[pos] = nums[i];
            pos+=2;
        }
        
        else {
            arr[neg] = nums[i];
            neg+=2;
        }
    }
    return arr;
};